### PR TITLE
Fix RPC name override

### PR DESCRIPTION
**Pull request by dhaidashenko**

### CODE REVIEW

File: core/chains/legacyevm/chain.go [hunk 1]
```diff
@@ -475,13 +475,19 @@ func newEthClientFromCfg(cfg evmconfig.NodePool, noNewHeadsThreshold time.Durati
 	for i, node := range nodes {
 		if node.SendOnly != nil && *node.SendOnly {
 			name := fmt.Sprintf("eth-sendonly-rpc-%d", i)
+			if node.Name != nil && *node.Name != "" {
+				name = *node.Name
+			}
 			rpc := evmclient.NewRPCClient(lggr, empty, (*url.URL)(node.HTTPURL), name, int32(i), chainID,
 				commonclient.Secondary)
```
Comment (jmank88) [inline core/chains/legacyevm/chain.go:482]: It is safe to pass the `nodeName` directly. We validate that all nodes are named. Notice that we already pas it without nil check on the line below. 
```suggestion
			rpc := evmclient.NewRPCClient(lggr, empty, (*url.URL)(node.HTTPURL), *node.Name, int32(i), chainID,
				commonclient.Secondary)
```

**Review by samsondav — COMMENTED** [review]
Can we get a test demonstrating what this fixes?

**Comment by jmank88** [comment]
> Can we get a test demonstrating what this fixes?

We haven't had the ability to test prom metrics coherently in the past, since they are all mixed up in global state. However, we could add a txtar test and checks the /metrics endpoint, like for /health: https://github.com/smartcontractkit/chainlink/blob/develop/testdata/scripts/health/multi-chain.txtar

**Comment by dhaidashenko** [comment]
> > Can we get a test demonstrating what this fixes?
> 
> We haven't had the ability to test prom metrics coherently in the past, since they are all mixed up in global state. However, we could add a txtar test and checks the /metrics endpoint, like for /health: https://github.com/smartcontractkit/chainlink/blob/develop/testdata/scripts/health/multi-chain.txtar

Added the test, but it does not feel that this approach is perfect. The script seems too complex for txtar test. Also it's quite limited in terms of available metrics as node does not perform any actions. Automated check against soaked node might be better a better option.

**Comment by jmank88** [comment]
> > > Can we get a test demonstrating what this fixes?
> > 
> > 
> > We haven't had the ability to test prom metrics coherently in the past, since they are all mixed up in global state. However, we could add a txtar test and checks the /metrics endpoint, like for /health: [`develop`/testdata/scripts/health/multi-chain.txtar](https://github.com/smartcontractkit/chainlink/blob/develop/testdata/scripts/health/multi-chain.txtar?rgh-link-date=2024-01-19T17%3A07%3A47Z)
> 
> Added the test, but it does not feel that this is the right approach. The script seems too complex for txtar test. Also it's quite limited in terms of available metrics as node does not perform any actions. Automated check against soaked node might be a better option.

I was hoping it could be made simpler, but soak tests would be fine too :shrug:

**Review by prashantkumar1982 — APPROVED** [review]
Thanks for fixing this!

**Comment by dhaidashenko** [comment]
> > > > Can we get a test demonstrating what this fixes?
> > > 
> > > 
> > > We haven't had the ability to test prom metrics coherently in the past, since they are all mixed up in global state. However, we could add a txtar test and checks the /metrics endpoint, like for /health: [`develop`/testdata/scripts/health/multi-chain.txtar](https://github.com/smartcontractkit/chainlink/blob/develop/testdata/scripts/health/multi-chain.txtar?rgh-link-date=2024-01-19T17%3A07%3A47Z)
> > 
> > 
> > Added the test, but it does not feel that this is the right approach. The script seems too complex for txtar test. Also it's quite limited in terms of available metrics as node does not perform any actions. Automated check against soaked node might be a better option.
> 
> I was hoping it could be made simpler, but soak tests would be fine too 🤷

@jmank88 on second thought, implementation of a more generalized "framework" for metrics testing seems like a high-effort low-priority task. As the current approach gets the job done in this particular case, do you mind merging the fix as is?

**Pull request merged**